broker/pkg/storage/redis/scripts.go

```diff
@@ -124,5 +130,69 @@ var scripts = struct {
 	
 	    return data
 	`,
-	delete: `return 100`,
+	getByConsumerId: `
+	    local consumer_key = KEYS[1]
+	    local time_reference_MS = KEYS[2]
+	
+	    local data = {}
+	
+	    local message_ids = redis.call("ZRANGEBYSCORE", consumer_key, time_reference_MS, "+inf")
+	    for idx, message_key in pairs(message_ids) do
+	        local message = redis.call("HMGET", message_key, "ID", "TimestampMS", "BodyID", "Body", "ProducerGroupID", "LockConsumerID", "BucketID", "Version")
+	        local tmp = {}
+			table.insert(tmp, message[1])
+			table.insert(tmp, message[2])
+			table.insert(tmp, message[3])
+			table.insert(tmp, message[4])
+			table.insert(tmp, message[5])
+			table.insert(tmp, message[6])
+			table.insert(tmp, message[7])
+			table.insert(tmp, message[8])
+			table.insert(data, tmp)
+	    end
+	
+	    return data
+	`,
+	consumerDelete: `
+	    local timeline_key = KEYS[1]
+	    local consumer_id = KEYS[2]
+	    local time_reference_MS = KEYS[3]
+	    local messages = {}
+	    local errors = {}
+	
+	    local function deleteMessage(bucket_id, message_id, version)
+	        local bucket_key = timeline_key .. "::" .. bucket_id
+	
+	        -- check if it have lease and lease is valid
+	        local score = redis.call("ZSCORE", timeline_key .. "::" .. consumer_id, bucket_key .. "::" .. message_id)
```

Comment:
is the mesage removed from the "consumer_key" structure as well?

Reply:
Yes, it is (redis.call("ZREM", timeline_key .. "::" .. consumer_id, bucket_key .. "::" .. message_id) line 181. Comment associated with code block was wrong.